decidim-surveys/app/permissions/decidim/surveys/admin/permissions.rb

```diff
@@ -20,6 +20,15 @@ def permissions
             when :index, :show, :export_response
               permission_action.allow!
             end
+          when :questionnaire_publish_answers
+            case permission_action.action
+            when :index, :update, :destroy
+              if context.fetch(:survey).allow_answers
+                disallow!
```

Comment:
I guess this should be like this ? just to be consistent with the rest of file ?
```suggestion
                permission_action.disallow!
```